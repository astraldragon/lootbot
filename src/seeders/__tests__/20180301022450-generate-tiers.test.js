jest.mock("dotenv", () => ({
  config: jest.fn("config").mockReturnValue(null)
}));

const migration = require("../20180301022450-generate-tiers");

describe("seeders/20180301022450-generate-tiers", () => {
  beforeEach(() => {
    jest.spyOn(Date, "now").mockImplementation(() => 1487076708000);
    process.env.GUILD = "1";

    this.queryInterface = {
      bulkInsert: jest.fn("createTable").mockReturnValue(true),
      bulkDelete: jest.fn("dropTable").mockReturnValue(true)
    };

    this.Sequelize = {
      Op: { or: "or" }
    };
  });

  afterEach(() => {
    delete process.env.GUILD;
  });

  it("should throw error when seeding if no guild is provided", () => {
    delete process.env.GUILD;

    expect(() => {
      migration.up(this.queryInterface);
    }).toThrowError("GUILD env variable required");
  });

  it("should seed when a guild is provided", () => {
    migration.up(this.queryInterface);
    expect(this.queryInterface.bulkInsert).toMatchSnapshot();
  });

  it("should throw error when rolling back if no guild is provided", () => {
    delete process.env.GUILD;

    expect(() => {
      migration.down(this.queryInterface);
    }).toThrowError("GUILD env variable required");
  });

  it("should rollback when a guild is provided", () => {
    migration.down(this.queryInterface, this.Sequelize);
    expect(this.queryInterface.bulkDelete).toMatchSnapshot();
  });
});
