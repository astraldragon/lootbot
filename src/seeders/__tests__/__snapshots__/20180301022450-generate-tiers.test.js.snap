// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seeders/20180301022450-generate-tiers should rollback when a guild is provided 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Tiers",
      Object {
        "guild": "1",
        "name": Object {
          "or": Array [
            "Common",
            "Uncommon",
            "Rare",
            "Legendary",
          ],
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": true,
    },
  ],
}
`;

exports[`seeders/20180301022450-generate-tiers should seed when a guild is provided 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Tiers",
      Array [
        Object {
          "color": 4737096,
          "createdAt": 2017-02-14T12:51:48.000Z,
          "guild": "1",
          "image": "https://media.discordapp.net/attachments/342426492371992576/417019496109441024/common.png",
          "luckyWeight": 40,
          "name": "Common",
          "updatedAt": 2017-02-14T12:51:48.000Z,
          "weight": 50,
        },
        Object {
          "color": 4171061,
          "createdAt": 2017-02-14T12:51:48.000Z,
          "guild": "1",
          "image": "https://media.discordapp.net/attachments/342426492371992576/417019515893710858/uncommon.png",
          "luckyWeight": 30,
          "name": "Uncommon",
          "updatedAt": 2017-02-14T12:51:48.000Z,
          "weight": 37,
        },
        Object {
          "color": 12209815,
          "createdAt": 2017-02-14T12:51:48.000Z,
          "guild": "1",
          "image": "https://media.discordapp.net/attachments/342426492371992576/417019531018633216/rare.png",
          "luckyWeight": 25,
          "name": "Rare",
          "updatedAt": 2017-02-14T12:51:48.000Z,
          "weight": 10,
        },
        Object {
          "color": 15155217,
          "createdAt": 2017-02-14T12:51:48.000Z,
          "guild": "1",
          "image": "https://media.discordapp.net/attachments/342426492371992576/417019544675024916/legendary.png",
          "luckyWeight": 5,
          "name": "Legendary",
          "updatedAt": 2017-02-14T12:51:48.000Z,
          "weight": 3,
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": true,
    },
  ],
}
`;
